tsdb/engine/tsm1/engine.go

```diff
@@ -1856,6 +1856,18 @@ func (e *Engine) WriteSnapshot() (err error) {
 		if err == nil {
 			log.Info("Snapshot for path written", zap.String("path", e.path), zap.Duration("duration", elapsed))
 		}
+
+		// cleanup temp files
+		e.Cache.mu.Lock()
+		defer e.Cache.mu.Unlock()
+
+		if e.Cache.stats.SnapshotCount == 0 {
+			err = e.cleanupTempShardSnapshots()
```

Comment:
@ydkn thanks for the contribution!

This call should be unnecessary if the code to clean these up is working correctly (which we believe it is on linux in 1.8.5, but is broken on Windows). Ideally the fix would be to do the cleanup properly on windows as well rather than taking a relatively expensive lock here.

One way we can get tmp files left over is here, though it should not be happening after the recent compaction changes: https://github.com/influxdata/influxdb/blob/master-1.x/tsdb/engine/tsm1/compact.go#L1072

Another way would be some kind of renaming problem here: https://github.com/influxdata/influxdb/blob/master-1.x/tsdb/engine/tsm1/file_store.go#L764 resulting in the tmp file being left.

@davidby-influx believes that there may be an OS-specific call required on Windows somewhere to make this work, e.g. similar to https://github.com/influxdata/influxdb/blob/master-1.x/services/meta/client.go#L1022

I don't think this PR can be merged in its present form, since it may impact performance due to taking this mutex. But we would welcome some more data on which of the above cases you are hitting or a PR that addressed the problem more surgically.